app/scripts/modules/core/src/pipeline/config/triggers/artifacts/custom/CustomArtifactEditor.tsx

```diff
@@ -1,82 +1,107 @@
 import * as React from 'react';
-import { isFinite } from 'lodash';
-import { IArtifact, IArtifactEditorProps } from 'core/domain';
 
-const input = (artifact: IArtifact, field: keyof IArtifact, onChange: (a: IArtifact) => void) => (
-  <input
-    type="text"
-    className="form-control input-sm"
+import { IArtifact, IArtifactEditorProps, IArtifactKindConfig, IPipeline } from 'core/domain';
+import { SpelText } from 'core/widgets';
+
+const input = (artifact: IArtifact, field: keyof IArtifact, pipeline: IPipeline, onChange: (a: IArtifact) => void) => (
+  <SpelText
+    placeholder={''}
     value={artifact[field] || ''}
-    onChange={e => onChange({ ...artifact, [field]: e.target.value })}
+    onChange={(value: string) =>
+      onChange({
+        ...artifact,
+        type: 'custom',
+        customKind: true,
+        [field]: value,
+      })
+    }
+    pipeline={pipeline}
+    docLink={false}
   />
 );
 
-const SingleColumnCustomArtifactEditor = (props: IArtifactEditorProps) => {
-  const { artifact, onChange, groupClassName } = props;
-  const labelColumns = isFinite(props.labelColumns) ? props.labelColumns : 2;
-  const fieldColumns = isFinite(props.fieldColumns) ? props.fieldColumns : 8;
-  const labelClassName = 'col-md-' + labelColumns;
-  const fieldClassName = 'col-md-' + fieldColumns;
-  const formGroupClasses = groupClassName != null ? groupClassName : 'form-group row';
-  return (
-    <div>
-      <div className={formGroupClasses}>
-        <label className={labelClassName + ' sm-label-right'}>Type</label>
-        <div className={fieldClassName}>{input(artifact, 'type', onChange)}</div>
-      </div>
-      <div className={formGroupClasses}>
-        <label className={labelClassName + ' sm-label-right'}>Name</label>
-        <div className={fieldClassName}>{input(artifact, 'name', onChange)}</div>
-      </div>
-      <div className={formGroupClasses}>
-        <label className={labelClassName + ' sm-label-right'}>Version</label>
-        <div className={fieldClassName}>{input(artifact, 'version', onChange)}</div>
-      </div>
-      <div className={formGroupClasses}>
-        <label className={labelClassName + ' sm-label-right'}>Location</label>
-        <div className={fieldClassName}>{input(artifact, 'location', onChange)}</div>
-      </div>
-      <div className={formGroupClasses}>
-        <label className={labelClassName + ' sm-label-right'}>Reference</label>
-        <div className={fieldClassName}>{input(artifact, 'reference', onChange)}</div>
-      </div>
+export const CustomMatch: IArtifactKindConfig = {
+  label: 'Custom',
+  typePattern: /UNMATCHABLE/,
+  description: 'A custom-defined artifact.',
+  key: 'custom',
+  isDefault: true,
+  isMatch: true,
```

Comment:
Custom Artifact is not currently selectable in the expected-artifact selector component in the config UI.

This artifact kind config needs a `customKind: true` field to workaround this issue.  It appears to have been lost in the rewrite from angular to react.

Reply:
Restored in 5ab4885a4.